WeatherPortal/weatherapp/views.py

```diff
@@ -22,12 +64,66 @@ def register(request):
             return redirect(reverse('weather_in_user_city'))
 
 
-def your_weather(request):
+def your_weather(request, add_new_location=False, chosen_location=0):
+    """
+        Connect with openweather api and show information about weather in choosen city in website
+        Saved added city for only logged user
+        Check validator that city exist and for logged user that city exist in user's saved cities
+    """
+    cities_locations = []
+    if request.method == 'POST':
+        form = CityForm(request.POST)
+        if form.is_valid():
+            new_add_city = form.cleaned_data['name']
+            get_location_for_new_city = get_city_location(new_add_city)
+            if len(get_location_for_new_city) == 1:
+                add_new_location = True
+
+                #If is one location only
+            else:
+                for counter_cities in range(len(get_location_for_new_city)):
+                    city_location = {
+                        'city': new_add_city,
+                        'lat': get_location_for_new_city[counter_cities]['lat'],
+                        'lon': get_location_for_new_city[counter_cities]['lon'],
+                        'country': get_location_for_new_city[counter_cities]['country'],
+                        'state': get_location_for_new_city[counter_cities]['state'],
+                        'number': counter_cities,
+                    }
+                    cities_locations.append(city_location)
+
+
+
+    if add_new_location:
+        print(f'user choose {chosen_location}')
+    form = CityForm()
+    context = {
+        'cities_locations': cities_locations,
+        'form': form,
+    }
+    return render(request, 'weatherapp/yourweather.html', context)
+
+
+def add_chosen_location(request, chosen_location):
+    return your_weather(request, add_new_location=True, chosen_location=chosen_location)
```

Comment:
Tutaj nowy endpoint zapewnia tylko `chosen_location`, równie dobrze mógłbyś to przesyłać pod `your_weather`